### DISCUSSION TITLE
Plugin developer question - is there a way to disable flow launchers auto rank system when result order matters

### DISCUSSION THREAD

**Discussion by Or1g3n** [Flow-Launcher/Flow.Launcher#2993] [Q&A] [answered]
Plugin developer question - is there a way to disable flow launchers auto rank system when result order matters

I maintain a plugin where order matters (e.g., help feature shows available commands that should be ordered alphabetically, there is a sort feature where user can specify sort order, etc.). The issue I have found is that flow launcher appears to keep a ranking of results in a UserSelectedRecord.json file and that this ranking takes precedence over all; I tried hard coding a sort order in the c# list object but the ranking by Flow still took precedence; clearing the data stored in the UserSelectedRecord.json resulted in expected sort order.

I can see why this auto ranking system can be very useful in most use cases given it avoids the user having to manually search multiple matches to select the most common result record, however, in for my use case where sort order matters, this is not the desired behavior.

Is there a way to turn this off using the plugin api?

**Comment by taooceros** [comment]
I guess no for now. Maybe you can just setting a crazy score difference for each results so the order would not be reordered unless clicked very very many time?

**Reply by Or1g3n** [reply]
Wow that answer was provided with lightning speed, didn't expect such a quick response, thanks @taooceros ! That is a good thought, I will give that idea a go if the ability to turn this behavior off for a given plugin is not available currently.

Other than the potential work around you suggested, do you think my request is valid as a feature request? If so I can put one in.

**Reply by Or1g3n** [reply]
@taooceros 
Appreciate your suggested work around. To ensure order I assigned the first result item a value of 1,000,000,000 and then incremented down in increments of 1,000,000 (user would have to select an item one million plus one to mess up desired order lol). After testing this appears to work.

For long term solution I suggested a flag to Results class as indicated in issue @jjw24 linked below.

**Reply by taooceros** [reply]
Good to see that it works. Yeah we should have some flag for that.

**Comment by jjw24** [comment, marked as answer]
https://github.com/Flow-Launcher/Flow.Launcher/issues/2998
